src/lib.rs

```diff
@@ -209,16 +209,27 @@ where
         self.radio_kind.do_tx(timeout_in_ms).await?;
         match self
             .radio_kind
-            .process_irq(self.radio_mode, self.rx_continuous, &mut self.delay, None, None)
+            .process_irq(
+                self.radio_mode,
+                self.rx_continuous,
+                TargetIrqState::Done,
+                &mut self.delay,
+                None,
+                None,
+            )
             .await
         {
-            Ok(()) => Ok(()),
+            Ok(TargetIrqState::Done) => {
+                self.radio_mode = RadioMode::Standby;
+                Ok(())
+            }
             Err(err) => {
                 self.radio_kind.ensure_ready(self.radio_mode).await?;
                 self.radio_kind.set_standby().await?;
                 self.radio_mode = RadioMode::Standby;
                 Err(err)
             }
+            Ok(_) => unreachable!(),
```

Comment:
I'm wondering if this is actually unreachable. What if you start an RX before, drop the future, and then try to TX? Couldn't the irq trigger with TargetIrqState::Preamble in that case? Feels safer if this was somewhat ignored or returned an error.

Reply:
process_irq will only return `TargetIrqState::PreambleReceived` if explicitly asked to. In this case it's asked to only return `TargetIrqState::Done`

Reply:
Looking again I see what you mean. Looks good to me then.

Reply:
I'm thinking about changing the API to be a little bit more type-stateful. That'd resolve some issues like device being left in an inconsistent state after cancellation of a future. Also, it'd remove confustion between tx interrupts and rx interrupts.

Reply:
It would be nice to have a proper chat about thr overall layout of this crate. If you have the time you could join us at https://matrix.to/#/!zyeeBoubKcHiwSJylm:matrix.org